v2/tools/generator/internal/functions/property_assignment_function_builder.go

```diff
@@ -0,0 +1,457 @@
+	for destinationName, destinationEndpoint := range destinationProperties {
+		if sourceEndpoint, ok := sourceProperties[destinationName]; ok {
+			err := assign(sourceEndpoint, destinationEndpoint)
+			if err != nil {
+				return errors.Wrapf(err, "assigning %s", destinationName)
+			}
+		}
+	}
+
+	return nil
+}
+
+// readPropertiesFromPropertyBag populates destination properties that don't have a matching source property by reading
+// the values from a property bag on the source object.
+// sourceEndpoints is a set of endpoints that can be read from.
+// destinationEndpoints is a set of endpoints that can be written to.
+// assign is a function that will be called for each matching property, with the source and destination endpoints
+// for that assignment.
+func (builder *PropertyAssignmentFunctionBuilder) readPropertiesFromPropertyBag(
+	sourceEndpoints conversions.ReadableConversionEndpointSet,
+	destinationEndpoints conversions.WritableConversionEndpointSet,
+	assign func(reader *conversions.ReadableConversionEndpoint, writer *conversions.WritableConversionEndpoint) error,
+) error {
+	prop := builder.findPropertyBagProperty(builder.sourceType())
+	if prop == nil {
+		// No property bag on our source type, nothing to read from
+		return nil
+	}
+
+	// for each destination property that doesn't already have a source endpoint, synthesize an endpoint that reads
+	// from the property bag
+	for destinationName, destinationEndpoint := range destinationEndpoints {
+		if _, ok := sourceEndpoints[destinationName]; ok {
+			// already have a source endpoint for this property
+			continue
+		}
+
+		// Create a new endpoint that reads from the property bag
+		sourceEndpoint := conversions.NewReadableConversionEndpointReadingPropertyBagMember(destinationName, destinationEndpoint.Endpoint().Type())
+		err := assign(sourceEndpoint, destinationEndpoint)
+		if err != nil {
+			return errors.Wrapf(err, "assigning %s from property bag", destinationName)
+		}
+
+		builder.readsFromPropertyBag = true
+	}
+
+	return nil
+}
+
+// writePropertiesToPropertyBag stores source properties that don't have a matching destination property by writing
+// the values to a property bag on the destination object.
+// sourceEndpoints is a set of endpoints that can be read from.
+// destinationEndpoints is a set of endpoints that can be written to.
+// assign is a function that will be called for each matching property, with the source and destination endpoints
+// for that assignment.
+func (builder *PropertyAssignmentFunctionBuilder) writePropertiesToPropertyBag(
+	sourceEndpoints conversions.ReadableConversionEndpointSet,
+	destinationEndpoints conversions.WritableConversionEndpointSet,
+	assign func(reader *conversions.ReadableConversionEndpoint, writer *conversions.WritableConversionEndpoint) error,
+) error {
+	prop := builder.findPropertyBagProperty(builder.destinationType())
+	if prop == nil {
+		// No property bag on our source type, nothing to read from
+		return nil
```

Comment:
Similar question as in that other location -- if we actually have a property we need to write to a property bag, and thus we call the `writePropertiesToPropertyBag` function, should that be an error?

Will the generated code be incorrect if not?

Reply:
This is unchanged code that was relocated. 

By definition of the way conversion methods are generated, if there's no property bag found on the destination type, there are no properties that need to be written to it.

If this was an error case, we'd be unable to generate conversions that populate API types based on the Storage variant.